File: 3869-smallest-index-with-digit-sum-equal-to-index/smallest-index-with-digit-sum-equal-to-index.py
class Solution(object):
    def smallestIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n=len(nums)
        for i in range(0,n):
            sum=0

            while nums[i]>0 :
             cnt=nums[i]%10
             sum+=cnt
             nums[i]//=10
            if(sum==i):
             return i
        return -1

        

File: 3869-smallest-index-with-digit-sum-equal-to-index/smallest-index-with-digit-sum-equal-to-index.cpp
class Solution {
public:
    int smallestIndex(vector<int>& nums) {
        int n=nums.size();
        for(int i=0;i<n;i++){
            int sum=0;
            while(nums[i]>0){
                int cnt=nums[i]%10;
                sum+=cnt;
                nums[i]/=10;
            }
            if(i==sum) return i;
        }
        return -1;
    }
};